_playwright-tests/UI/TemplateCRUD.spec.ts

```diff
@@ -0,0 +1,115 @@
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+      const row = await getRowByNameOrUrl(page, `${repoName}`);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+    await test.step('Navigate to templates, ensure the Add content template button can be clicked', async () => {
+      await navigateToTemplates(page);
+      await closePopupsIfExist(page);
+      const AddButton = page.locator('button#createContentTemplateButton');
+      // Wait for the Add button to become enabled (up to 10 seconds)
+      await AddButton.first().isEnabled({ timeout: 10000 });
+    });
+    await test.step('Create a template', async () => {
+      await page.getByRole('button', { name: 'Add content template' }).click();
+      await page.getByRole('button', { name: 'Select architecture' }).click();
+      await page.getByRole('option', { name: 'aarch64' }).click();
+      await page.getByRole('button', { name: 'Select version' }).click();
+      await page.getByRole('option', { name: 'el9' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      const rowRHELRepo = await getRowByNameOrUrl(page, smallRHRepo);
+      await rowRHELRepo.getByRole('checkbox', { name: 'Select row' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      const rowRepo = await getRowByNameOrUrl(page, repoName);
+      await rowRepo.getByRole('checkbox', { name: 'Select row' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByText('Use latest content', { exact: true }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      // "add template modal"
+      await page.getByText('add template modal', { exact: true });
+      await page.getByPlaceholder('Enter name').fill(`${templateName}`);
+      await page.getByPlaceholder('Description').fill('Template test');
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Create other options' }).click();
+      await page.getByText('Create template only', { exact: true }).click();
+    });
+    await test.step('Read and update values in the template', async () => {
+      await navigateToTemplates(page);
+      await closePopupsIfExist(page);
+      const rowTemplate = await getRowByNameOrUrl(page, templateName);
+      await rowTemplate.getByRole('button', { name: templateName }).click();
+      await expect(page.getByLabel('Breadcrumb').first()).toHaveText('RHELContentTemplates');
+      await page.getByLabel('Breadcrumb').nth(2);
```

Comment:
Hmm, You should use some action here, e.g. you want to wait for it to be visible?

Reply:
Yes, I will change that.